Date for case count tooltip

theswerd opened the pull request:
### Date for Tooltip

Makes the tooltip for case count show the date rather than the case amount

Add the Issue number(s) assigned to you that this PR fully resolves, if any
Closes #1575 #1460
#### Screenshots
<details>
<summary>Screenshots</summary>
<img width="520" alt="Screen Shot 2020-10-10 at 2 47 51 PM" src="https://user-images.githubusercontent.com/38309438/95665697-a83a4600-0b07-11eb-948e-614d7dc27678.png">

</details>



#### How did you test the change?
* [x] iOS Simulator
* [ ] Android Emulator
* [ ] iOS Device
* [ ] Android Device
* [ ] `curl` to a dev App Engine server

<!-- FILL OUT THE CHECKLIST BELOW -->

---

#### Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).

- [x] REQUIRED: Do you have an Issue **assigned to you** for this PR?
- [x] Provided detailed pull request description and a succinct title
- [x] Tested your changes, especially after any code review iterations.
- [x] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [x] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [docs/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/docs/credits.yaml) file (if you want it to be).

advayDev1 commented on client/flutter/lib/components/recent_numbers_graph.dart:
```diff
@@ -33,8 +34,26 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
+              getTooltipItem: (barChartGroupData, _, barRodData, index) {
+            final date = DateTime.fromMillisecondsSinceEpoch(
+                widget.timeseries[index].epochMsec.toInt());
+            final formattedDate = DateFormat('M/d/y').format(date);
```
this format is wrong for most of the non-US world

brunobowden replied:
@theswerd - see this earlier comment from me about using "October 6, 2020" as the format to avoid localization issues: https://github.com/WorldHealthOrganization/app/issues/1575#issuecomment-706651820

advayDev1 replied:
as long as you force en_US somehow as the locale you should be fine, but the date for the month names has to be loaded in dynamically otherwise.  See https://pub.dev/documentation/intl/latest/intl/DateFormat-class.html

advayDev1 replied:
to be more specific, i think disabling the localizations forces the language part of the tag to be 'en' but I am unclear whether v1 would launch with 'en_NG' or 'en_US' unless someone has tested this system in nigeria yet

brunobowden replied:
We're reaching out to the NCDC. They'll be involved with the next round of testing to check out things like this.

theswerd commented:
@advayDev1 while I believe I used the mechanisms correctly can you confirm that i am pulling the right date for each point?

theswerd commented:
New date format per @brunobowden's suggestion
<img width="520" alt="Screen Shot 2020-10-13 at 12 50 08 AM" src="https://user-images.githubusercontent.com/38309438/95831638-22c4ba80-0cee-11eb-862c-ed80ce2240b7.png">

brunobowden reviewed (approved):
@theswerd - we'll have Nigeria check this out during QA. It looks good to me, though I don't have a perfect understanding on the localization.

@Mohit020888 - once this lands, try it out in India and see if you notice any issues around this and the other UI changes that have been happening.

brunobowden commented:
@theswerd - please also squash your commits before merging

brunobowden commented:
@theswerd - by creating new PR #1615, it means that it loses the history and discussion that we just had. I'd suggest reopening this PR and then doing the squash client side. It should be straight forward:
```
git fetch upstream
git rebase -i upstream/master
# edit all the "pick" entries except for the first one to "s" to squash
# hopefully no conflicts but follow the process if needed
# edit the commit messages to create a single combined message
git push -f origin
# this will force update the repo: "theswerd/app"
```
At this point, you can enjoy a simple squashed commit. See GitHub website or GitHub Desktop client along the way to check it's as expected.

brunobowden commented:
@theswerd - sorry this has been a long process. I just noticed that this will lose the count number that was there before. It should also show "X,XXX Cases" and "XXX Deaths" underneath the date. This is what https://covid19.who.int already shows per country (see screenshots below). Don't bother adding the "Daily Increase" and "Daily Change" as that's too much for a small mobile screen.

Having the Data and Count both displayed will mean you can check the data. Pick out a spike (or the right most entry) and then check that the case count, date and highlighted column match up with the WHO Covid-19 dashboard.

Sorry this has been a long process. If you prefer to do this in another PR, I can merge in this change and make it a two-step process. Let me know.

WHO Covid-19 Dashboard:

<img width="274" alt="Screen Shot 2020-10-17 at 11 43 23 AM" src="https://user-images.githubusercontent.com/679339/96351201-c73c5900-106e-11eb-9e09-0625d9c3f61e.png">

<img width="336" alt="Screen Shot 2020-10-17 at 11 43 10 AM" src="https://user-images.githubusercontent.com/679339/96351207-ca374980-106e-11eb-8966-2d36c24149af.png">

theswerd commented:
BarTooltipItem has the following props:

PropName | Description | default value
-- | -- | --
text | text string of each row in the tooltip bubble | null
textStyle | TextStyle of the showing text row | null

We will be unable to do more than one text color in the tooltip.

With this in mind should I contact the product team to see if we can come up with a new design, or should I fork the fl_charts repo and create a change to allow children as widgets rather than text? @brunobowden @yukuairoy

brunobowden commented:
@theswerd - I should've been more explicit. The colour doesn't matter.... just the Date and Count. As long as those can be done across two lines in black text. Then that's fine.

theswerd commented:
<img width="463" alt="Screen Shot 2020-10-18 at 12 57 30 PM" src="https://user-images.githubusercontent.com/38309438/96378570-b4df1f80-1141-11eb-8c49-47c89c42bc03.png">
<img width="463" alt="Screen Shot 2020-10-18 at 12 57 18 PM" src="https://user-images.githubusercontent.com/38309438/96378572-b6a8e300-1141-11eb-8389-30b9409cc3b5.png">
- Added Numbers
- Fixed Tooltip getting cut off problem
Closes #1460

brunobowden commented on client/lib/components/recent_numbers_graph.dart:
```diff
@@ -33,8 +33,36 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
+              final formattedDate = DateFormat('MMMM d, yyyy').format(date);
+
+              final commaRegEx = RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))');
+              final formattedCount =
```
Great engineering is a study in elegance and efficiency. In other words, I can understand what the regex does with a lot of effort that shouldn't be needed. Comma separators for large numbers is a common enough activity that there will be a standard library for it. As above, this also has the benefit of getting localization for free too.
```suggestion
              final formattedCount = NumberFormat().format(barRodData.y);
```

brunobowden reviewed (changes requested):
See how the suggested simplifications give you what's needed:

![Screenshot_1603060710](https://user-images.githubusercontent.com/679339/96387741-3db58580-1159-11eb-88e0-d046b4c70636.png)
